client/app/components/account_config.coffee

```diff
@@ -0,0 +1,382 @@
+        'accountType'
+    ]
+
+    _mailboxesFields: [
+        'id'
+        'mailboxes'
+        'favoriteMailboxes'
+        'draftMailbox'
+        'sentMailbox'
+        'trashMailbox'
+    ]
+
+
+    _accountSchema:
+        properties:
+            label: allowEmpty: false
+            name: allowEmpty: false
+            login: allowEmpty: false
+            password: allowEmpty: false
+            imapServer: allowEmpty: false
+            imapPort: allowEmpty: false
+            imapSSL: allowEmpty: true
+            imapTLS: allowEmpty: true
+            smtpServer: allowEmpty: false
+            smtpPort: allowEmpty: false
+            smtpSSL: allowEmpty: true
+            smtpTLS: allowEmpty: true
+            smtpLogin: allowEmpty: true
+            smtpMethod: allowEmpty: true
+            smtpPassword: allowEmpty: true
+            draftMailbox: allowEmpty: true
+            sentMailbox: allowEmpty: true
+            trashMailbox: allowEmpty: true
+            accountType: allowEmpty: true
+
+
+    getInitialState: ->
+        return @accountToState null
+
+
+    # Do not update component if nothing has changed.
+    shouldComponentUpdate: (nextProps, nextState) ->
+        isNextState = _.isEqual nextState, @state
+        isNextProps = _.isEqual nextProps, @props
+        return not (isNextState and isNextProps)
+
+
+    componentWillReceiveProps: (props) ->
+        # prevents the form from changing during submission
+        if props.selectedAccount? and not props.isWaiting
+            @setState @accountToState props
+
+        else
+
+            if props.error?
+
+                if props.error.name is 'AccountConfigError'
+                    errors = {}
+                    field = props.error.field
+
+                    if field is 'auth'
+                        errors.login = t 'config error auth'
+                        errors.password = t 'config error auth'
+
+                    else
+                        errors[field] = t 'config error ' + field
+
+                    @setState errors: errors
+            else
+
+                if not props.isWaiting and not _.isEqual(props, @props)
+                    @setState @accountToState null
+
+
+    render: ->
+        mainOptions = @buildMainOptions()
+        mailboxesOptions = @buildMailboxesOptions()
+        titleLabel = @buildTitleLabel()
+        tabParams = @buildTabParams()
+
+        Container
+            id: 'mailbox-config'
+            key: "account-config-#{@props.selectedAccount?.get 'id'}"
+        ,
+            Title text: titleLabel
+            if @props.tab?
+                Tabs tabs: tabParams
+            if not @props.tab or @props.tab is 'account'
+                AccountConfigMain mainOptions
+            else
+                AccountConfigMailboxes mailboxesOptions
+
+
+    # Ask to main layout manager to display error as notification toasters.
+    renderError: (errors) ->
+        LayoutActionCreator.alertError t 'account errors'
```

Comment:
This function has no meaning with your refactoring, you could use `LayoutActionCreator.alertError` directly in the validation method. Or if you really want to keep it, rename it, so it doesn't confuse with an actual sub render function.